### PR TITLE
`qmat` switch

### PR DESCRIPTION
**Pull request by tlunet**
After implementing most of the core coefficients generation of `pySDC` into [`qmat`](https://github.com/Parallel-in-Time/qmat), here is the associated PR finalizing the switch mentioned in #436.

Main changes :

- `LagrangeApproximation` and `NodeGenerator` classes (and their associated tests) are now in `qmat`
- the base `CollBase` class is now mostly an interface to the `Collocation` class from `qmat`
- the base `Sweeper` class produces implicit and explicit QDelta matrices using `qmat` generators
- additional base `pip` dependency for `pySDC` set to `qmat>=0.1.8`
- docstring / documentation update to link to `qmat` documentation when needed

Additional changes :

- includes complete adaptation to numpy v2, based on the first changes done by @brownbaerchen in his fork. The problematic behavior for the `parallelSDC` project has been patch by removing testing (and coverage) of the plotting function for `preconditioner_playground_MPI`. If this PR is accepted, no need to merge #443 anymore as it was already merged in this PR.
- `Sweeper.get_qDelta_implicit` and `Sweeper.get_qDelta_explicit` only have one argument now : `qd_type`. The previous `coll` one was simply replaced by `self.coll`, as no `Sweeper` instance in `pySDC` code use another `Collocation` class as the base underlying collocation for the sweeper. This was a weird mix between OOP and Procedural Programming that does not make sense but rather the code more complex than it should be.
- complete renaming of the core sub-modules to follow a naming consistent with contribution guidelines, using the refactoring tools from VSCode. Only one issue had to be solved manually for a private class name based attribute in a `Hook` class (see https://github.com/tlunet/pySDC/commit/32f64bf032119f989c9302c3c077d1cb4fc07de8, I won't even try to understand the idea behind this sort of attribute ...)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -30,8 +28,15 @@ dependencies = [
     'sympy>=1.0',
     'numba>=0.35',
     'dill>=0.2.6',
+    'qmat',
```
Comment (pancetta) [inline pyproject.toml:31]: I guess that should have `>=0.1.8` here, too?

**Comment by pancetta** [comment]
Great work, thanks a lot. Taking the quadrature stuff out of pySDC is a very good idea and I appreciate your work.

**Comment by pancetta** [comment]
Not sure why you're obscuring this PR with so much renaming, though.. this makes it much harder to see what actually happens here.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -9,6 +9,11 @@ Others are specific conventions chosen for the pySDC library, that may follow Py
 Finally, while `pySDC` provides many base functionalities that implement classical flavors of SDC, it also allows problem-specific applications through Object-Oriented Programming (OOP) and the implementation of custom inherited classes.
 This follows a specific OOP framework, you can look at the page on [custom implementations](./docs/contrib/04_custom_implementations.md) for more details. Additional guideline are also given on how to [document the code](./docs/contrib/05_documenting_code.md) in `pySDC`.
 
+> 📣 Some core components (collocation matrix, $Q-\Delta$ coefficients, Lagrange barycentric interpolation) 
+> are currently implemented and developed in the [`qmat` companion package](https://github.com/Parallel-in-Time/qmat).
```
Comment (pancetta) [inline CONTRIBUTING.md:13]: I'd leave out the `currently`, because this sounds like this is a temporary thing.

**Comment by pancetta** [comment]
Is there a reason why the CI pipeline is back to 40+minutes now? Or is that a GitHub glitch?  I see no obvious cause.

File: pyproject.toml (hunk 1 above)
Comment (tlunet) [reply pyproject.toml:31]: Done in the next commit

**Comment by tlunet** [comment]
> Not sure why you're obscuring this PR with so much renaming, though.. this makes it much harder to see what actually happens here.

Well, this was a big change anyway, and the naming of the core classes has been confusing me since a long time ... so that's mostly shooting two birds with one stone.

**Comment by pancetta** [comment]
> > Not sure why you're obscuring this PR with so much renaming, though.. this makes it much harder to see what actually happens here.
> 
> Well, this was a big change anyway, and the naming of the core classes has been confusing me since a long time ... so that's mostly shooting two birds with one stone.

Understood. As a preference, I think [smaller PRs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/best-practices-for-pull-requests#write-small-prs) with a clear focus are better, though.

**Comment by pancetta** [comment]
And no, this is not stopping me from merging this, of course!

File: CONTRIBUTING.md (hunk 2 above)
Comment (tlunet) [reply CONTRIBUTING.md]: Done

**Comment by tlunet** [comment]
> And no, this is not stopping me from merging this, of course!

I guessed :sweat_smile: ... but don't worry, I'll stick to smaller PR with one dedicated topic from now.

Also in my opinion, it would make sense to drop a new release after the PR is merged.

**Comment by pancetta** [comment]
For sure and already on my todo list!

**Comment by tlunet** [comment]
> Is there a reason why the CI pipeline is back to 40+minutes now? Or is that a GitHub glitch? I see no obvious cause.

Probably a GitHub glitch ... but project testing still take quite some time (especially the `parallelSDC`, `DAE` and `Resilience`). Where are we on testing projects only for one Python version ?

**Comment by pancetta** [comment]
Unless this will significantly shorten the CI time, I'd rather have broad coverage.

**Comment by tlunet** [comment]
> Unless this will significantly shorten the CI time

Probably not, since most are run in parallel ... just significantly less computation ressource used (mostly energy saving)

**Pull request merged**